### ISSUE TITLE
C error with optional array return type

### ISSUE THREAD

**Issue by bstnbuck** [vlang/v#8628]
**V version:** V 0.2.2 b8d93df
**OS:** Windows 10

**What did you do?**
Compiled this code :
```v 
import os

fn make_files_list(dir string)?[]string{
	dir_content := os.ls(dir) ?
	mut files := []string{}
	for content in dir_content{
		if is_dir(content){
			files << make_files_list(content)?
		}else{
			files << content
		}
	}
	return files
}

fn main(){
	println(make_files_list("testfiles"))
}
```

**What did you expect to see?**
Works?
 
**What did you see instead?**
C error like this:
```
Microsoft (R) C/C++-Optimierungscompiler Version 19.28.29336 für x64
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten.

cl -w /we4013 /volatile:ms /Fo"C:\Users\buckt\AppData\Local\Temp\v\test.7368734097747192153.tmp.c.obj" /MDd /D_DEBUG "C:\Users\buckt\AppData\Local\Temp\v\test.7368734097747192153.tmp.c" -I "D:\Windows Kits\10\Include\10.0.18362.0\ucrt" -I "D:\Programme\Visual Studio\VC\Tools\MSVC\14.28.29333\include" -I "D:\Windows Kits\10\Include\10.0.18362.0\um" -I "D:\Windows Kits\10\Include\10.0.18362.0\shared" kernel32.lib user32.lib advapi32.lib dbghelp.lib /link /NOLOGO /OUT:"D:\buckt\Desktop\vshred\test.exe" /LIBPATH:"D:\Windows Kits\10\Lib\10.0.18362.0\ucrt\X64" /LIBPATH:"D:\Windows Kits\10\Lib\10.0.18362.0\um\X64" /LIBPATH:"D:\Programme\Visual Studio\VC\Tools\MSVC\14.28.29333\lib\X64" /DEBUG:FULL

test.7368734097747192153.tmp.c
C:\Users\buckt\AppData\Local\Temp\v\test.7368734097747192153.tmp.c(12715): error C2440: "Initialisierung": "Option_array_string" kann nicht in "array_string" konvertiert werden
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

**Comment by bstnbuck** [comment]
This code compiles without problems:
```v
import os

fn make_files_list(dir string)[]string{
	dir_content := os.ls(dir) or {return []} //added this here
	mut files := []string{}
	for content in dir_content{
		if is_dir(content){
			files << make_files_list(content)
		}else{
			files << content
		}
	}
	return files
}

fn main(){
	println(make_files_list("testfiles"))
}
```

**Comment by spaceface777** [comment]
Fixed in #9021.
